### PR TITLE
[Cases] Fixing types

### PR DESCRIPTION
**Pull request by jonathan-buttner**
Running the type checker succeeds. I'm seeing this error though on the integration tests:

```
   └-> "after all" hook in "cases security and spaces enabled: trial"
     │ debg deleting space: space1
     └- ✖ fail: cases security and spaces enabled: trial "after all" hook in "cases security and spaces enabled: trial"
     │      Error: Expected status code of 204, received 404 Not Found: { statusCode: 404, error: 'Not Found', message: 'Not Found' }
     │       at SpacesService.delete (x-pack/test/common/services/spaces.ts:43:15)
     │       at runMicrotasks (<anonymous>)
     │       at processTicksAndRejections (internal/process/task_queues.js:93:5)
     │       at deleteSpaces (x-pack/test/case_api_integration/common/lib/authentication/index.ts:49:5)
     │       at deleteSpacesAndUsers (x-pack/test/case_api_integration/common/lib/authentication/index.ts:70:3)
     │       at Context.<anonymous> (x-pack/test/case_api_integration/security_and_spaces/tests/trial/index.ts:22:7)
     │       at Object.apply (node_modules/@kbn/test/src/functional_test_runner/lib/mocha/wrap_function.js:73:16)
     │
     │
```

I suppose we could wrap the call in a `try/catch` and just ignore the error since it seems like the space doesn't exist for some reason. Maybe it was accidentally deleted by a test file somewhere?

### CODE REVIEW

File: x-pack/plugins/cases/server/client/mocks.ts [hunk 1]
```diff
@@ -5,115 +5,81 @@
...
+export interface CasesClientMock extends CasesClient {
+  cases: CasesSubClientMock;
+  attachments: AttachmentsSubClientMock;
+  userActions: UserActionsSubClientMock;
+  subCases: SubCasesClientMock;
+}
+
+export const createCasesClientMock = (): CasesClientMock => {
+  const client: PublicContract<CasesClient> = {
+    casesClientInternal: (jest.fn() as unknown) as CasesClientInternalMock,
```
Comment (jonathan-buttner) [inline x-pack/plugins/cases/server/client/mocks.ts:67]: When my PR gets merged we can remove this.

**Comment by cnasikas** [comment]
> Running the type checker succeeds. I'm seeing this error though on the integration tests:
> 
> ```
>    └-> "after all" hook in "cases security and spaces enabled: trial"
>      │ debg deleting space: space1
>      └- ✖ fail: cases security and spaces enabled: trial "after all" hook in "cases security and spaces enabled: trial"
>      │      Error: Expected status code of 204, received 404 Not Found: { statusCode: 404, error: 'Not Found', message: 'Not Found' }
>      │       at SpacesService.delete (x-pack/test/common/services/spaces.ts:43:15)
>      │       at runMicrotasks (<anonymous>)
>      │       at processTicksAndRejections (internal/process/task_queues.js:93:5)
>      │       at deleteSpaces (x-pack/test/case_api_integration/common/lib/authentication/index.ts:49:5)
>      │       at deleteSpacesAndUsers (x-pack/test/case_api_integration/common/lib/authentication/index.ts:70:3)
>      │       at Context.<anonymous> (x-pack/test/case_api_integration/security_and_spaces/tests/trial/index.ts:22:7)
>      │       at Object.apply (node_modules/@kbn/test/src/functional_test_runner/lib/mocha/wrap_function.js:73:16)
>      │
>      │
> ```
> 
> I suppose we could wrap the call in a `try/catch` and just ignore the error since it seems like the space doesn't exist for some reason. Maybe it was accidentally deleted by a test file somewhere?

Let's pair to investigate why this is happening. It should be there!

File: x-pack/plugins/cases/server/client/mocks.ts (hunk 1 above)
Comment (cnasikas) [reply x-pack/plugins/cases/server/client/mocks.ts:67]: And we can also add the `stats` subclient and the `getTags`, `getReporters` methods to the `CasesSubClientMock`

**Review by cnasikas — APPROVED** [review]
LGTM! We should add the `stats` sub client and the `getTags`, `getReporters` methods to the `CasesSubClientMock` when https://github.com/elastic/kibana/pull/96507 is ready.

**Pull request merged**